chrome-extension/popup.js

```diff
@@ -127,37 +129,125 @@ function downloadButtonCallback(chapter_list_url) {
 var download_count = 0;
 var expected_download_count = 0;
 
+
+function getBookNameArray(urls)
+{
+  var names = "";
+
+  for (var i = 0 ; i < urls.length ; i++)
+    names += i+1 + ". " + getBookNameFromUrl(urls[i]) + "\n";
+
+  return names;
+}
+
+function isInQuenePage(url)
+{
+  var re = new RegExp('(\.+://www.safaribooksonline.com/s/)\.*');
+  var match = re.exec(url);
+
+  var re2 = new RegExp('(\.+://www.safaribooksonline.com/s/\\?limit=100)\.*');
+  var match2 = re2.exec(url);
+
+  if(match != null && match2 != null)
+    return 2;     // return 2 if the page has loaded 100 books in the Safari Queue page
+  else if (match != null)
+    return 1;     // return 1 if the page is in Safari Queue page
+  else 
+    return 0;     // return 0 if the page is NOT in Safari Queue page
+}
+
+function clickedEvent(url)
```

Comment:
I extracted these code to a new function because I used this function twice(see line 222 & 247), I do think that we should use a function to contain these code but ofcoz a better function name is needed.